aopy/analysis.py

```diff
@@ -856,3 +854,282 @@ def linear_fit_analysis2D(xdata, ydata, weights=None, fit_intercept=True):
+    npts = altcond_train.shape[0]
+    ntrials = nullcond_test.shape[1]
+
+    # Get max accLLR value for threshold calculation by calculating the max accLLR with the training data
+    accLLR, _ = calc_activity_onset_accLLR(altcond_train, altcond_train, nullcond_train, modality=modality, bin_width=bin_width)
+    max_accLLR = np.max(accLLR)
+    
+    thresh_props = np.arange(0, 1, thresh_step_size)
+    fa_rates = []    
+    for tp in thresh_props:
+        accLLR, selection_time_idx = calc_activity_onset_accLLR(nullcond_test, altcond_train, nullcond_train, modality, bin_width, thresh_proportion=tp, max_accLLR=max_accLLR, trial_average=False) 
+        print('maxaccllr', accLLR)
+        accLLR_altcond_within_time = np.sum(np.logical_and(accLLR > 0, selection_time_idx < npts), axis=0)
+        n_accllr_altcond_within_time = np.count_nonzero(accLLR_altcond_within_time)
+        false_alarms = n_accllr_altcond_within_time / ntrials
+        fa_rates.append(false_alarms)
+    fa_rates = np.array(fa_rates)
+    fa_rates_above_desired = thresh_props[fa_rates > false_alarm_prob]
+    best_tp = np.nan
+    if any(fa_rates_above_desired):
+        best_tp = fa_rates_above_desired[0]
+    return best_tp, thresh_props, fa_rates
+
+
+def accLLR_wrapper(data_altcond, data_nullcond, modality, bin_width, train_prop_input=0.7, thresh_step_size=0.01, false_alarm_prob=0.05, trial_average=True, match_selectivity=True):
+    '''
+    1. Separates data into 'model building' (training + test) and 'implementing' groups by trials 
+        Assumes model building dataset is split into a 60/40 split
+    2. Calculates accLLR threshold on training and test data
+    3. Implements accLLR on 'implementing' data
+    
+    Data allocation breakdown by trial (train_prop_input = 0.7)
+    40% is used to train threshold proportion (train)
+    30% is used to validate threshold proportion (test)
+    30% is used to calculate selection time (validate)
+
+
+    Args:
+        data_altcond (npts, nch, ntrials):
+        data_nullcond (npts, nch, ntrials):
+        modality (str): either 'spikes' or 'lfp'
+        train_prop_input (float): proportion of trials to build the model with
+        
+    Returns:
+        (tuple): Tuple containing:
+            | **accllr_altcond (nt, nch):**
+            | **accllr_nullcond (nt, nch):**
+            | **selection_time_altcond (nch):**
+            | **selection_time_nullcond (nch):** 
+    '''
+    nt = data_altcond.shape[0]
+    nch = data_altcond.shape[1]
+    ntrials = data_altcond.shape[2]
+
+    # Split into model building (train, validate) and testing data sets (outputs (ntrial, npt) datasets)
+    # test_data_altcond, build_data_altcond, test_data_nullcond, build_data_nullcond = model_selection.train_test_split(data_altcond.T, data_nullcond.T, test_size=train_prop_input)
+    ntrain_trials = int(np.ceil(ntrials*0.4))
+    nvalid_trials = int(np.ceil(ntrials*0.3))
+    ntest_trials = ntrials - ntrain_trials - nvalid_trials
```

Comment:
currently this is hard-coded. we can do something like:
```suggestion
    ntrain_trials = int(np.ceil(ntrials*train_prop_input*0.7))
    nvalid_trials = int(np.ceil(ntrials*train_prop_input*0.3))
    ntest_trials = ntrials - ntrain_trials - nvalid_trials
```
so at least we have flexibility over the amount of data being used for training vs testing